### PR TITLE
Refactor: Start phasing out `first_round` and create session in `get_block`

### PR DESCRIPTION
**Pull request by drikusroor**
This pull request includes a refactoring of the `get_block` function to phase out the `first_round` methods and create a session directly in the `get_block` function. If the `first_round` does not exist anymore in a block's rules file, it will call `next_round` directly. The frontend has also been updated to take the session_id from the get_block response. `get_block` will still call `first_round` if it does exist on the block's rules file in question. This allows us to phase it out over time and prevent this PR from becoming a monster to review.

Additionally, tests have been fixed and added after the refactoring of the `first_round` methods and session creation in `get_block`.

Finally, the `Block` and `Trial` component have been partially configured to TypeScript and some types have been updated/added to accomodate this.

Related to #973

### CODE REVIEW

File: backend/session/models.py [hunk 1]
```diff
@@ -75,15 +80,25 @@ def load_json_data(self):
...
 
+    def save_json_data_key(self, key: str, value):
+        """Set json data key"""
+        data = self.load_json_data()
+        data[key] = value
+        self.save_json_data(data)
+
+    def load_json_data_key(self, key: str, default=None):
+        """Get json data key"""
+        return self.load_json_data()[key] if key in self.load_json_data() else default
```
Comment (BeritJanssen) [inline backend/session/models.py:91]:
```suggestion
        return self.load_json_data().get(key, default)
```

File: backend/session/models.py [hunk 2]
```diff
@@ -75,15 +80,25 @@ def load_json_data(self):
         """Get json data as object"""
         return self.json_data if self.json_data else {}
 
+    def save_json_data_key(self, key: str, value):
```
Comment (BeritJanssen) [inline backend/session/models.py:83]: What are these two methods used for?

**Review by BeritJanssen — APPROVED** [review]
Nice to do this in steps, indeed. Only puzzed by the `save / load_json_data_key` functions, which I couldn't find being used.

File: backend/session/models.py (hunk 2 above)
Comment (drikusroor) [reply backend/session/models.py:83]: I was using them, but then I think I didn't need them anymore so I'll remove the methods as well here.

**Pull request merged**